piccolo/apps/migrations/auto/schema_differ.py

```diff
@@ -291,21 +284,17 @@ def rename_tables(self) -> AlterStatements:
     def _get_snapshot_table(
         self, table_class_name: str
     ) -> t.Optional[DiffableTable]:
-        snapshot_table = self.schema_snapshot_map.get(table_class_name, None)
-        if snapshot_table:
+        if snapshot_table := self.schema_snapshot_map.get(
+            table_class_name, None
+        ):
             return snapshot_table
-        else:
-            if (
+        if table_class_name in self.rename_tables_collection.new_class_names:
+            class_name = self.rename_tables_collection.renamed_from(
                 table_class_name
-                in self.rename_tables_collection.new_class_names
-            ):
-                class_name = self.rename_tables_collection.renamed_from(
-                    table_class_name
-                )
-                snapshot_table = self.schema_snapshot_map.get(class_name)
-                if snapshot_table:
-                    snapshot_table.class_name = table_class_name
-                    return snapshot_table
+            )
+            if snapshot_table := self.schema_snapshot_map.get(class_name):
+                snapshot_table.class_name = table_class_name
+                return snapshot_table
```

Comment:
This looks great, but we can't use the walrus operator at the moment because of Python 3.7 😞

Reply:
Yes, that was the issue, I was thinking to propose the drop of py3.7 but I guess it's better to wait until the new release of py3.11 and see if it's way much to drop the support on py3.7

Reply:
It has just over a year of life left: https://endoflife.date/python

There's an argument to remove it, but I can imagine some people being locked to Python 3.7 for weird corporate reasons.

Reply:
> It has just over a year of life left: https://endoflife.date/python
> 
> There's an argument to remove it, but I can imagine some people being locked to Python 3.7 for weird corporate reasons.

Yeah same happen to me in some projects I work in with my old company, we use py3.7 and py3.6 in the testing, and its caused much of issues while refactoring the code and fix some bugs 🐛